pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -621,31 +621,49 @@ func (c *Controller) getProxyServiceInstancesFromMetadata(proxy *model.Proxy) ([
 		if !f {
 			return nil, fmt.Errorf("failed to find model service for %v", hostname)
 		}
+
+		tps := make(map[model.Port]*model.Port)
 		for _, port := range svc.Spec.Ports {
 			svcPort, f := modelService.Ports.Get(port.Name)
 			if !f {
 				return nil, fmt.Errorf("failed to get svc port for %v", port.Name)
 			}
-			targetPort, err := findPortFromMetadata(port, proxy.Metadata.PodPorts)
+			portNum, err := findPortFromMetadata(port, proxy.Metadata.PodPorts)
 			if err != nil {
 				return nil, fmt.Errorf("failed to find target port for %v: %v", proxy.ID, err)
 			}
-			// Construct the ServiceInstance
-			out = append(out, &model.ServiceInstance{
-				Service:     modelService,
-				ServicePort: svcPort,
-				Endpoint: &model.IstioEndpoint{
-					Address:         proxy.IPAddresses[0],
-					EndpointPort:    uint32(targetPort),
-					ServicePortName: svcPort.Name,
-					// Kubernetes service will only have a single instance of labels, and we return early if there are no labels.
-					Labels:         proxy.WorkloadLabels[0],
-					ServiceAccount: svcAccount,
-					Network:        c.endpointNetwork(proxy.IPAddresses[0]),
-					Locality:       util.LocalityToString(proxy.Locality),
-					Attributes:     model.ServiceAttributes{Name: svc.Name, Namespace: svc.Namespace},
-				},
-			})
+			// Dedupe the target ports here - Service might have configured multiple ports to the same target port,
+			// we will have to create only one ingress listener per port and protocol so that we do not endup
+			// complaining about listener conflicts.
+			targetPort := model.Port{
+				Port:     portNum,
+				Protocol: svcPort.Protocol,
```

Comment:
```
- name: http-7442
   port: 7442
   protocol: TCP
   targetPort: 7442
- name: tcp-8443
   port: 8443
   protocol: TCP
   targetPort: 7442
```
What's the output for the above?

Reply:
I think you need to get the converted svc port's protocol

Reply:
Here is the output of the above - It generates two service instances
```
Service Instance 1 
Service Port :  {http-7442 7442 HTTP} 
Service Endpoint : {app=test-app tcp 128.0.0.1 http-7442 kubernetes://pod1.nsa <nil> spiffe://cluster.local/ns/nsa/sa/foo   7442 0 { svc1 nsa  map[] map[]} disabled} 
Service Instance 2 
Service Port :  {tcp-8443 8443 TCP} 
Service Endpoint : {app=test-app tcp 128.0.0.1 tcp-8443 kubernetes://pod1.nsa <nil> spiffe://cluster.local/ns/nsa/sa/foo   7442 0 { svc1 nsa  map[] map[]} disabled} 
```

I have a test case "single proxy ip multiple ports" that covers this scenario. 

Am I missing some thing?

Reply:
Oh, this port is already converted.